packages/core/src/config/parse.ts

```diff
@@ -171,14 +198,507 @@ export const assertValidUserConfigFields = (config: UserChugSplashConfig) => {
   }
 }
 
+/**
+ * Parses and validates the elements of an array. This function is used whenever the encoding of
+ * the array is `inplace` (for fixed size arrays) or `dynamic_array`, but not `bytes`, which is
+ * used for dynamic bytes and strings. Works recursively with the `parseAndValidateVariable` function.
+ *
+ * @param array Array to parse and validate.
+ * @param storageObj Solidity compiler JSON output describing the layout for this array.
+ * @param storageTypes Full list of storage types allowed.
+ * @param nestedSlotOffset Not used, only included here because of the shared recursiveLayoutIterator structure.
+ * @returns Array with it's elements converted into the correct type for the parsed chugsplash config.
+ */
+export const parseAndValidateArrayElements = (
+  array: any[],
+  storageObj: SolidityStorageObj,
+  storageTypes: {
+    [name: string]: SolidityStorageType
+  },
+  nestedSlotOffset: string,
+  dereferencer: ASTDereferencer
+): Array<any> => {
+  const elementType = getStorageType(
+    storageObj.type,
+    storageTypes,
+    dereferencer
+  ).base
+
+  if (elementType === undefined) {
+    throw new Error(
+      `Could not encode array elements for: ${storageObj.label}. Please report this error to the developers, this should never happen.`
+    )
+  }
+
+  // Arrays always start at a new storage slot with an offset of zero.
+  const bytesOffset = 0
+
+  // Iterate over the array and encode each element in it.
+  const parsedArray: any[] = []
+  for (const element of array) {
+    parsedArray.push(
+      parseAndValidateVariable(
+        element,
+        {
+          astId: storageObj.astId,
+          contract: storageObj.contract,
+          label: storageObj.label,
+          offset: bytesOffset,
+          slot: '0',
+          type: elementType,
+        },
+        storageTypes,
+        nestedSlotOffset,
+        dereferencer
+      )
+    )
+  }
+  return parsedArray
+}
+
+/**
+ * Handles parsing and validating fixed-size arrays
+ *
+ * @param props standard VariableHandler props. See ./iterator.ts for more information.
+ * @returns
+ */
+export const parseInplaceArray: VariableHandler<ParsedConfigVariable> = (
```

Comment:
I think this should either be `parseAndValidateInplaceArray` or the call below should be `parseArrayElements` for consistency

Reply:
What other functions are you referring too?

Reply:
Did we decide whether to make everything `parseAndValidate`, `parse`, or keep as is?